src/config/config.ts

```diff
@@ -118,71 +122,70 @@ export class Config extends ConfigFile<ConfigFile.Options> {
-          },
+      },
+      { key: Config.DEFAULT_DEV_HUB_USERNAME },
+      { key: Config.DEFAULT_USERNAME },
+      {
+        key: Config.ISV_DEBUGGER_SID,
+        encrypted: true,
+        input: {
+          // If a value is provided validate it otherwise no value is unset.
+          validator: (value) => value == null || isString(value),
+          failedMessage: Config.messages.getMessage('InvalidIsvDebuggerSid'),
         },
-        {
-          key: Config.ISV_DEBUGGER_URL,
-          input: {
-            // If a value is provided validate it otherwise no value is unset.
-            validator: (value) => value == null || isString(value),
-            failedMessage: Config.messages.getMessage('InvalidIsvDebuggerUrl'),
-          },
+      },
+      {
+        key: Config.ISV_DEBUGGER_URL,
+        input: {
+          // If a value is provided validate it otherwise no value is unset.
+          validator: (value) => value == null || isString(value),
+          failedMessage: Config.messages.getMessage('InvalidIsvDebuggerUrl'),
         },
-        {
-          key: Config.DISABLE_TELEMETRY,
-          input: {
-            validator: (value) => value == null || ['true', 'false'].includes(value.toString()),
-            failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue'),
-          },
+      },
+      {
+        key: Config.DISABLE_TELEMETRY,
+        input: {
+          validator: (value) => value == null || ['true', 'false'].includes(value.toString()),
+          failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue'),
         },
-        // This should be brought in by a plugin, but there isn't a way to do that right now.
-        {
-          key: 'restDeploy',
-          hidden: true,
-          input: {
-            validator: (value) => value != null && ['true', 'false'].includes(value.toString()),
-            failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue'),
-          },
+      },
+      // This should be brought in by a plugin, but there isn't a way to do that right now.
+      {
+        key: 'restDeploy',
+        hidden: true,
+        input: {
+          validator: (value) => value != null && ['true', 'false'].includes(value.toString()),
+          failedMessage: Config.messages.getMessage('InvalidBooleanConfigValue'),
         },
-        {
-          key: Config.MAX_QUERY_LIMIT,
-          input: {
-            validator: (value) => isNumber(value),
-            failedMessage: Config.messages.getMessage('InvalidNumberConfigValue'),
-          },
+      },
+      {
+        key: Config.MAX_QUERY_LIMIT,
+        input: {
+          validator: (value) => isNumber(value),
+          failedMessage: Config.messages.getMessage('InvalidNumberConfigValue'),
         },
-      ];
-    }
+      },
+    ];
+
+    Config.allowedProperties = [...(Config.allowedProperties || []), ...defaultConfigPropertyMetas];
```

Comment:
Takes any external `allowedProperties` that have been set and adds the internal defaults, and sets the combined result on `allowedProperties`.

Reply:
This seems a little weird to me. Consumers are not supposed to call the constructor directly, but there is a global and local config instance, which means that `allowedProperties` would have the defaults twice?